messaging-client-databus/src/main/java/com/inmobi/databus/readers/LocalStreamCollectorReader.java

```diff
@@ -281,4 +285,77 @@ protected Date getTimeStampFromCollectorStreamFile(FileStatus file) {
     }
     return null;
   }
+
+  protected Long doRecursiveSizing(Path dir, PathFilter pathFilter) throws IOException {
+    Long pendingSize = 0l;
+    FileStatus[] fileStatuses = fsListFileStatus(dir, pathFilter);
+    if (fileStatuses == null || fileStatuses.length == 0) {
+      LOG.debug("No files in directory:" + dir);
+    } else {
+      for (FileStatus file : fileStatuses) {
+        if (file.isDir()) {
+          pendingSize += doRecursiveSizing(file.getPath(), pathFilter);
+        } else {
+          try {
+            Date currentTimeStamp = LocalStreamCollectorReader.
+                    getDateFromStreamFile(streamName, file.getPath().getName());
+            if (stopTime != null && stopTime.before(currentTimeStamp)) {
+              continue;
+            } else {
+              pendingSize += file.getLen();
+            }
+          } catch (Exception e) {
+            LOG.error("Exception while getting time from File " + file.getPath().toString(), e);
+            throw new IOException(e);
+          }
+        }
+      }
+    }
+    return pendingSize;
+  }
+
+  public Long getPendingSize() throws IOException {
+    Long pendingSize = 0L;
+    Calendar current = Calendar.getInstance();
+    Date now = current.getTime();
+    try {
+      current.setTime(getTimeStampFromMinutelyDirectory(getCurrentFile()));
```

Comment:
@sreedishps As mentioned earlier, there are many static helper methods around minDirFormatStr. E.g. 

public static Date getDateFromStreamDir(Path streamDir, Path dir) {
    String pathStr = dir.toString();
    int startIndex = streamDir.toString().length() + 1;
    String dirString = pathStr.substring(startIndex,
        startIndex + minDirFormatStr.length());
    try {
      return minDirFormat.get().parse(dirString);
    } catch (ParseException e) {
      LOG.warn("Could not get date from directory passed", e);
      throw new IllegalArgumentException(e);
    }
  }

This directly returns the Date object. You could check its existing usages. Is it sufficient for your use-case?